AznBoy00/soen343team5
# Unit of Work applied to the Cart

#### What it does:  
This PR applies the Unit of Work object relational behavioral pattern to loan/cart. When an item is added to the cart the UOW sets the dirtybit to true and addes them to the UOW array via the IMAP. Once the users clicks on checkout items in the UOW array are checked for dirty bits, and verified against items in cart to determine whether or not its a loan action. This is where the query will occur. @AznBoy00 will implement the actualy DB query in #97.  

#### How to test:  
In order to test, log in as a regular user `u2@biblio.ca` then add items to cart and click on checkout. Once you click on checkout you will see the output in the console.  

Resolves #125

Authors : @kkhh001, @kcamcam, @KyCoding
